src/checks/extra_blank_line.rs

```diff
@@ -0,0 +1,88 @@
+use crate::checks::Check;
+use crate::common::*;
+
+pub(crate) struct ExtraBlankLineChecker {
+    template: &'static str,
+    name: &'static str,
+    blank_counter: usize,
+}
+
+impl ExtraBlankLineChecker {
+    fn message(&self) -> String {
+        return format!("{}: {}", self.name, self.template);
+    }
+}
+
+impl Default for ExtraBlankLineChecker {
+    fn default() -> Self {
+        Self {
+            name: "ExtraBlankLine",
+            template: "Extra blank line detected",
+            blank_counter: 0,
+        }
+    }
+}
+
+impl Check for ExtraBlankLineChecker {
+    fn run(&mut self, line: &LineEntry) -> Option<Warning> {
+        self.blank_counter = if line.is_empty() {
+            self.blank_counter + 1
+        } else {
+            0
+        };
+
+        if self.blank_counter > 1 {
```

Comment:
I think you should use `HashSet<usize>` and store there a line number instead of this solution. For example:
```rust
pub(crate) struct ExtraBlankLineChecker {
    name: &'static str,
    line_numbers: HashSet<usize>,
    template: &'static str,
}

impl Default for ExtraBlankLineChecker {
    fn default() -> Self {
        Self {
            name: "ExtraBlankLineChecker",
            line_numbers: HashSet::new(),
            template: "Extra blank line detected",
        }
    }
}

impl Check for ExtraBlankLineChecker {
    fn run(&mut self, line: &LineEntry) -> Option<Warning> {
        if !line.is_empty() {
            return None;
        }

        self.line_numbers.insert(line.number);

        if self.line_numbers.contains((line.number - 1).borrow()) {
            return Some(Warning::new(line.clone(), self.message()));
        }

        None
    }
}
```

Reply:
I think it will solve the problem with comments.

Reply:
This approach will definitely work, but it is more complicated.
Are you sure, that we will don't check comments in the future?
If so, I will implement this solution.

Reply:
Yes, I'm sure that we will not check comments in the future.

Reply:
I used `last_blank_number: Option<usize>` instead `HashSet<usize>` but the idea is same.